tests/test_room_api.py

```diff
@@ -204,23 +208,48 @@ def test_with_options_hls(self, room_api: RoomApi):
             target_window_duration=None,
         )
         properties.additional_properties = {"s3": None}
-
         component_hls = ComponentHLS(id=component.id, type="hls", properties=properties)
 
+        assert response == component_hls
         assert component == component_hls
 
     def test_with_options_rtsp(self, room_api: RoomApi):
         _, room = room_api.create_room(video_codec=CODEC_H264)
 
-        room_api.add_component(room.id, options=RTSP_OPTIONS)
+        response = room_api.add_component(room.id, options=RTSP_OPTIONS)
         component = room_api.get_room(room.id).components[0]
 
         component_rtsp = ComponentRTSP(
-            id=component.id, type="rtsp", properties=ComponentPropertiesRTSP()
+            id=component.id,
+            type="rtsp",
+            properties=ComponentPropertiesRTSP(
+                source_uri=RTSP_OPTIONS.source_uri,
+                keep_alive_interval=RTSP_OPTIONS.keep_alive_interval,
+                reconnect_delay=RTSP_OPTIONS.reconnect_delay,
+                rtp_port=RTSP_OPTIONS.rtp_port,
+                pierce_nat=RTSP_OPTIONS.pierce_nat,
+            ),
         )
 
+        assert response == component_rtsp
         assert component == component_rtsp
 
+    def test_with_options_file(self, room_api: RoomApi):
```

Comment:
All test components look similar. Could we extract it to a common function and use a table-driven test pattern  (it is mostly used in golang)?

Reply:
Thanks for the resources, this is a good idea.

Reply:
Because the File Component test is excluded it's getting complicated to make a table-driven test. However, I refactored the tests using helper function so it looks a bit cleaner. Let me know what you think.